src/sipp.cpp

```diff
@@ -1334,6 +1334,38 @@ static void setup_media_sockets()
     }
 }
 
+/* Seed the random number generator from a combination of
+ * - Seconds since epoch
+ * - Process ID
+ * - Hostname
+ */
+void randomseed(void)
+{
+    time_t       t             = time(nullptr);
+    pid_t        p             = getpid();
+    unsigned int seed          = t;
+    char         hostname[256] = { 0 };
+    int          index;
+
+    if (p != 0) {
+        // We don't XOR with the PID because if we launch twice successively
+        // it's possible the PID increment could precisely counter the time
+        // increment.
+        seed *= p;
+    }
+
+    // Might truncate, or might leave 0s after the end.
+    gethostname(hostname, 256);
+
+    for (index = 0; index < 256; index++) {
```

Comment:
I also could have done `gethostname(.., 255)` and iterated here while `hostname[index] != '\0'`, but I didn't see much in it either way.